### DISCUSSION TITLE
is @sveltejs/enhanced-img a good match for docs sites with a bunch of images?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [sveltejs/kit#11043] [Q&A]
Hi @benmccann  👋

Thanks for a great project, the `enhanced-img`.

We did a small POC on using this with our docs site written in Sveltekit. Here we have around 100 images, and in development, it works great. But the build time was dramatically increased.

Is `enhanced-img` a good fit for such a use case - or is it meant to be used with fewer images?

A few notes:

- On a Mac Studio the build time was increased from around 1 minute, to 2 minutes 
- On a default github action, the build time was increased by 10 minutes (!)

**Comment by david-plugge** [comment]
Seems kind of crazy, 10 minutes for 100 images?
I don't know a lot about github actions, but maybe there is a way to cache the generated images so this process only needs to be executed once per image unless you change something.

**Comment by benmccann** [comment]
Hi, this is something I've given some thought to, but haven't encountered myself yet or had any time to address. A couple ideas:
- https://github.com/JonasKruckenberg/imagetools/issues/522
- https://github.com/JonasKruckenberg/imagetools/issues/635

Help with those issues would be very much welcomed!

If your code happens to be public, I'd be interested to see an example of a site where it's really slow today.

**Reply by alexbjorlig** [reply]
At the moment the code is not public. Our team will take a look at issues, but to be honest we will probably fallback to uplaodcare for the moment right now

**Comment by benmccann** [comment]
@alexbjorlig can you share some more details about your case:
- how many CPUs do you have on your GitHub action?
- is there a typical image size that you have in your project?
- are you using `sizes` on many of your images?

**Reply by alexbjorlig** [reply]
1. We use the standard Github hosted runner
2. PNG screenshots, typically around 250 - 750 KB
3. No - when we experimented it was just the "default" :)

**Comment by benmccann** [comment]
Making a note of a related issue: https://github.com/sveltejs/kit/issues/11120

**Comment by benmccann** [comment]
Caching has been added in `vite-imagetools`. https://github.com/sveltejs/kit/pull/12055 will upgrade to the latest version to enable caching by default. Hopefully that addresses the issue for everyone
